### PR TITLE
Allow dropping a column used by an old but not currrent partition spec

### PR DESCRIPTION
**Pull request by islamismailov**
Adding another unit test to make sure that columns used by an old but not currrent partition spec are allowed to be dropped.

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/TestPartitionSpecInfo.java [hunk 1]
```diff
@@ -85,6 +85,31 @@ public void testSpecInfoPartitionedTable() {
...
+    table.updateSchema().deleteColumn("id").commit();
+
+    Assert.assertEquals(newSpec, table.spec());
+    Assert.assertEquals(newSpec, table.specs().get(newSpec.specId()));
+    Assert.assertEquals(spec, table.specs().get(spec.specId()));
+    Assert.assertEquals(
+        ImmutableMap.of(spec.specId(), spec, newSpec.specId(), newSpec), table.specs());
+    Assert.assertNull(table.specs().get(Integer.MAX_VALUE));
+    Assert.assertEquals(
+        "Schema must have one less column", initialColSize - 1, table.schema().columns().size());
```
Comment (rdblue) [inline core/src/test/java/org/apache/iceberg/TestPartitionSpecInfo.java:110]: This check could be better if you produced the expected schema and compared. You can use equality comparison for the underlying struct type. Just compare `table.schema().asStruct()`.

Comment (islamismailov) [reply core/src/test/java/org/apache/iceberg/TestPartitionSpecInfo.java:110]: will do

**Comment by islamismailov** [comment]
Can somebody stamp this? It's only adding a unit test so is super safe

**Comment by islamismailov** [comment]
(i did address Ryan's feedback)

**Review by Fokko — APPROVED** [review]
This looks good. I added similar tests to Spark earlier, but I think it is better to have this in core (as well). Thanks @islamismailov

**Pull request merged**